internal/cmd/connect/command_custom_plugin_create.go

```diff
@@ -0,0 +1,83 @@
+package connect
+
+import (
+	pcmd "github.com/confluentinc/cli/internal/pkg/cmd"
+	"github.com/confluentinc/cli/internal/pkg/errors"
+	"github.com/confluentinc/cli/internal/pkg/examples"
+	"github.com/confluentinc/cli/internal/pkg/output"
+	"github.com/spf13/cobra"
+	"path/filepath"
+)
+
+type pluginCreateOut struct {
+	Id         string `human:"ID" serialized:"id"`
+	Name       string `human:"Name" serialized:"name"`
+	ErrorTrace string `human:"Error Trace,omitempty" serialized:"error_trace,omitempty"`
+}
+
+func (c *customPluginCommand) newCreateCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:         "create <name>",
+		Short:       "Create a custom connector plugin.",
+		Args:        cobra.ExactArgs(1),
+		RunE:        c.createCustomPlugin,
+		Annotations: map[string]string{pcmd.RunRequirement: pcmd.RequireNonAPIKeyCloudLogin},
+		Example: examples.BuildExampleString(
+			examples.Example{
+				Text: "Create a custom connector plugin ",
+				Code: "confluent connect custom-plugin create plugin_name --config-file config.json",
+			},
+		),
+	}
+
+	cmd.Flags().String("plugin-file", "", "ZIP/JAR plugin file.")
+	cmd.Flags().String("description", "", "description of plugin")
+	cmd.Flags().String("documentation-link", "", "document link of plugin")
+	cmd.Flags().String("connector-class", "", "connector class of custom plugin")
+	cmd.Flags().String("connector-type", "", "connector type of custom plugin")
+	cmd.Flags().String("sensitive-properties", "", "sensitive properties config of custom plugin")
+
+	cobra.CheckErr(cmd.MarkFlagRequired("connector-class"))
+	cobra.CheckErr(cmd.MarkFlagRequired("connector-type"))
+
+	cobra.CheckErr(cmd.MarkFlagFilename("plugin-file", "zip", "jar"))
+	cobra.CheckErr(cmd.MarkFlagRequired("plugin-file"))
+	return cmd
+}
+
+func (c *customPluginCommand) createCustomPlugin(cmd *cobra.Command, args []string) error {
+	displayName := args[0]
+	description, err := cmd.Flags().GetString("description")
+	documentationLink, err := cmd.Flags().GetString("documentation-link")
+	pluginFileName, err := cmd.Flags().GetString("plugin-file")
+	connectorClass, err := cmd.Flags().GetString("connector-class")
+	connectorType, err := cmd.Flags().GetString("connector-type")
+	sensitivePropertiesString, err := cmd.Flags().GetString("sensitive-properties")
+
+	extension := filepath.Ext(pluginFileName)[1:]
+	if extension != "zip" && extension != "jar" {
+		return errors.Errorf(errors.PluginInvalidExtensionErrorMsg)
+	}
+
+	resp, err := c.V2Client.GetPresignedUrl(extension)
+	if err != nil {
+		return err
+	}
+
+	err = uploadFile(resp.GetUploadUrl(), pluginFileName, resp.GetUploadFormData())
+	if err != nil {
```

Comment:
Reducing the scope of local variables can help prevent bugs
```suggestion
	if err := uploadFile(resp.GetUploadUrl(), pluginFileName, resp.GetUploadFormData()); err != nil {
```